source/Core/Threads/GUIThread.cpp

```diff
@@ -996,6 +1000,40 @@ void startGUITask(void const *argument) {
   }
 #endif
 #endif
+  // al_qu Calibrate Cold Junction Compensation directly at boot, before internal components get warm.
+  uint32_t Temp = TipThermoModel::getTipInC();
+  if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0) {
```

Comment:
@Ralim
First of all:
Do you think this approach makes sense?

And if so, do you have time to look into this?
Especially this line here:
```C#
if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0)
```